src/App.tsx

```diff
@@ -15,58 +15,73 @@ export const goods = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <main className="section container">
-    <h1 className="title">No goods selected</h1>
+type Good = (typeof goods)[number];
 
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-      {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-      <button data-cy="ClearButton" type="button" className="delete ml-3" />
-    </h1>
+export const App: React.FC = () => {
+  const [selectedGood, setSelectedGoods] = useState<Good | null>('Jam');
+  const selectedHandler = (good: Good | null) => {
+    setSelectedGoods(good);
+  };
 
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+  const title = selectedGood
+    ? `${selectedGood} is selected`
+    : 'No goods selected';
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+  const addButton = (good: Good) => (
+    <button
+      data-cy="AddButton"
+      type="button"
+      className="button"
+      onClick={() => selectedHandler(good)}
+    >
+      +
+    </button>
+  );
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  const removeButton = (
+    <button
+      data-cy="RemoveButton"
+      type="button"
+      className="button is-info"
+      onClick={() => selectedHandler(null)}
+    >
+      -
+    </button>
+  );
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {title}
+        {selectedGood && (
+          <button
+            data-cy="ClearButton"
+            type="button"
+            className="delete ml-3"
+            onClick={() => selectedHandler(null)}
+          />
+        )}
+      </h1>
 
-        <tr data-cy="Good">
-          <td>
-            <button data-cy="AddButton" type="button" className="button">
-              +
-            </button>
-          </td>
+      <table className="table">
+        <tbody>
+          {goods.map(good => (
+            <tr
+              data-cy="Good"
+              key={good}
+              className={
+                selectedGood === good ? 'has-background-success-light' : ''
```

Comment:
you dont need to pass empty className and use ternary, 
you can do this like that, 'className': condition